domain/cached_repo_request_driven.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import threading
+from copy import deepcopy
+from datetime import datetime, timedelta
+
+from domain.model.mesh_results import MeshResults
+from domain.repo import Repo
+from domain.types import TestID
+
+logger = logging.getLogger(__name__)
+
+
+class CachedRepoRequestDriven:
```

Comment:
Github doesn't show the diff :(

Reply:
> Github doesn't show the diff :(
This is a new file, entire file is shown

Reply:
I know, but the implementation is based on cached_repo.py